hermes-console-vue/src/composables/topic/use-topic/useTopic.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  fetchTopic as getTopic,
+  fetchTopicMessagesPreview as getTopicMessagesPreview,
+  fetchTopicMetrics as getTopicMetrics,
+  fetchTopicOwner as getTopicOwner,
+  fetchTopicSubscriptionDetails as getTopicSubscriptionDetails,
+  fetchTopicSubscriptions as getTopicSubscriptions,
+} from '@/api/hermes-client';
+import { ref } from 'vue';
+import type {
+  MessagePreview,
+  TopicMetrics,
+  TopicWithSchema,
+} from '@/api/topic';
+import type { Owner } from '@/api/owner';
+import type { Subscription } from '@/api/subscription';
+import type { UseTopic, UseTopicErrors } from '@/composables/types';
+
+export function useTopic(topicName: string): UseTopic {
```

Comment:
We will have to refactor existing composables to use similar approach to be consistent in whole codebase (in different task obviously)